wiki/Upgrading-Target-Sdk-Guide.md

```diff
@@ -0,0 +1,148 @@
+
+Once all fixed issues are verified, leave a follow-up comment on the upgrade target SDK tracking issue mentioning that verification has concluded and there are no remaining issues found.
+
+### Part 5: Submitting the upgrade
+
+The actual code change to upgrade the app comes in two parts:
+1. Upgrading the compile version (which may already be done as the team sometimes needs to update this for other reasons).
+2. Upgrading the runtime target SDK version.
+
+The sub-sections below detail each of the code changes needed to perform these upgrades.
+
+**Important caveats and notes**:
+- Both version upgrades can be done together in the same PR, but if they are split up the compile-time change (part (5.1) below) must happen first.
+- #5222 is an example of a PR that performs both steps in one, though it includes a few additional code changes that were needed as a result of the compile SDK change.
+- Please note the CI results for this upgrade change. Any failures are likely problems that will need to be fixed within the upgrade PR (if small, e.g. the change in #5222) or filed as a separate bug that will need to be fixed before the upgrade can be submitted (see part (4) above).
+- Updating the compile-time SDK version may require updating the build tools version. This is **not** a simple change and may cause difficult-to-fix breakages due to subtle compatibility issues between third-party dependencies and the build system configurations. If you suspect a build tools version upgrade is needed, please file an issue to track it and contact the developer workflow team lead to discuss next steps.
+- Updating tests to use a newer version of Android can be exceptionally complicated, and thus this is considered a completely separate exercise from upgrading production code. See part (6) below for more specifics.
+
+#### Part 5.1: Updating the compiled SDK version
+
+This code change essentially requires replacing the old SDK version number (e.g. 31) with the new one (e.g. 33), but only for compile-time behaviors. All needed changes are detailed below:
+1. [``.github/actions/set-up-android-bazel-build-environment/action.yml``](https://github.com/oppia/oppia-android/blob/dfb9a301280b9a46526cb2f5ca6329532fec6bf0/.github/actions/set-up-android-bazel-build-environment/action.yml#L75-L78) needs to be updated to install the correct SDK version (via the ``sdkmanager --install`` command).
+2. Bazel [``build_vars.bzl``](https://github.com/oppia/oppia-android/blob/dfb9a301280b9a46526cb2f5ca6329532fec6bf0/build_vars.bzl#L1) changes to ``BUILD_SDK_VERSION``.
+3. Gradle ``compileSdkVersion`` changes (e.g. for [``app/build.gradle``](https://github.com/oppia/oppia-android/blob/dfb9a301280b9a46526cb2f5ca6329532fec6bf0/app/build.gradle#L8)). Note that all module ``.gradle`` files will need to be updated in this way.
+
+#### Part 5.2: Updating the target SDK version
```

Comment:
Would the WORKSPACE file also require an update in this case? (apologies if I overlooked)

Reply:
Good question--it shouldn't anymore now that we've moved relevant build configurations to build_vars.bzl. It does still need to be updated for the target SDK, but that'll go away soonish (once we remove the APKs targets--they are rarely used and we should prefer using the AAB builds even for local development).

Reply:
Ah, thanks for the clarification!